app/walletButton.tsx

```diff
@@ -9,128 +9,177 @@ import {
-    setWallet(wallet);
-  }
+  const connectedWallets = Object.values(wallets).filter(Boolean).length;
 
   return (
     <>
       <a
         className="font-mono uppercase tracking-widest p-4 border-r text-sm leading-6 bg-primary cursor-pointer"
         onClick={onClick}
       >
-        {wallet ? formatAddress(wallet.address) : "CONNECT WALLET"}
+        {connectedWallets > 0 ? "WALLETS" : "CONNECT WALLET"}
       </a>
-      {modal && modal.type == "pickChain" ? (
-        <Modal
-          title="Connect Wallet"
-          onClose={() => setModal(undefined)}
-          style={{ maxWidth: 360 }}
-        >
-          <Button
-            className="w-full flex items-center text-left mt-2"
-            onClick={onConnectBitcoin}
-          >
-            <Image
-              src="/logo-btc.svg"
-              className="icon mr-2"
-              alt="Bitcoin Logo"
-              height={24}
-              width={24}
-            />{" "}
-            Bitcoin
-          </Button>
-          <Button
-            className="w-full flex items-center text-left mt-2"
-            onClick={onConnectEthereum}
-          >
-            <Image
-              src="/logo-eth.svg"
-              className="icon mr-2"
-              alt="Bitcoin Logo"
-              height={24}
-              width={24}
-            />{" "}
-            Ethereum
-          </Button>
-        </Modal>
-      ) : null}
       {modal && modal.type == "wallet" ? (
         <ModalWallet
-          wallet={wallet!}
           onClose={() => setModal(undefined)}
-          setWallet={setWallet}
         />
       ) : null}
     </>
   );
 }
 
 function ModalWallet({
-  wallet,
   onClose,
-  setWallet,
 }: {
-  wallet: Wallet;
   onClose: () => void;
-  setWallet: (_: undefined | object) => void;
 }) {
-  const [balance, setBalance] = useState(0);
+  const [wallets, setWallets] = useAtom(atomWallet);
+  const { address: ethAddress, isConnected: isEthConnected } = useAccount();
+  const { disconnect } = useDisconnect();
+  const { data: ethBalance } = useBalance({ address: ethAddress });
+  const { openConnectModal } = useConnectModal();
+
+  const [btcBalance, setBtcBalance] = useState(0);
 
   useEffect(() => {
-    (async () => {
-      setBalance(await wallet.getBalance());
-    })();
-  }, [wallet]);
+    async function fetchBitcoinBalance() {
+      if (wallets.bitcoin) {
+        const balance = await wallets.bitcoin.getBalance();
+        setBtcBalance(balance);
+      }
+    }
+    fetchBitcoinBalance();
+  }, [wallets.bitcoin]);
 
-  function onDisconnect() {
-    onClose();
-    setWallet(undefined);
+  const updateEthereumWallet = useCallback(() => {
+    if (isEthConnected && ethAddress) {
+      setWallets(prev => ({
+        ...prev,
+        ethereum: {
+          chain: "ethereum",
+          symbol: "ETH",
+          address: ethAddress,
+          getBalance: async () => Number(ethBalance?.formatted || 0),
+        },
+      }));
+    } else {
```

Comment:
prefer not using an else statement here. add a `return` on the previous line and unindent L81-85